tests/scripts/thread-cert/test_zero_len_external_route.py

```diff
@@ -111,7 +111,7 @@ def test(self):
         router2.remove_prefix('fd00:1234::/64')
         router2.add_prefix('fd00:5678::/64', 'paros')
         router2.register_netdata()
-        self.simulator.go(5)
+        self.simulator.go(305)
```

Comment:
Not clear why an increase is needed here; there is no change in IPv6 address of the DUT here, right? The DUT should get new netdata in about ~5 sec and handle the flags correctly.

Reply:
I agree. We should not need the increased wait time in this case.
- `leader` should be able to ping the address using its new preferred address based on the newly added `fd00:5678::/64` prefix as the source address with destination then matching the  `::/0` route in netdata 
- So ping message delivered to `router2`.

If I recall correctly, the test was failing without the increase, so somehow still having a deprecated address based on `fd00:1234::/64` seem to be causing an issue here (I think/expect when selecting source, we would select preferred address over non-preferred one). I will try to investigate to see why it was failing.

Reply:
I think I found the cause. There may be an issue with `Ip6::SelectSource()` that can cause a non-preferred address to be selected over a preferred one (depending on the order as we go through the addresses). Will submit a fix PR for this soon.

Another intresting observation is that routing within mesh may not work as we expect if the node uses its deprecated address as the source address due to how we apply Domain ID (in prefix TLV) when checking the route entries in netdata. I will file an issue with more details.

Reply:
@abtink with #9634 merged, is it already possible to change back the wait time from 305 -> 5 ?